naming/src/main/java/com/alibaba/nacos/naming/push/PushService.java

```diff
@@ -540,6 +541,7 @@ public void refresh() {
             Loggers.PUSH.error("[NACOS-PUSH] failed to push data: {} to client: {}, error: {}",
                 ackEntry.data, ackEntry.origin.getAddress().getHostAddress(), e);
             ackMap.remove(ackEntry.key);
+            udpSendTimeMap.remove(ackEntry.key);
```

Comment:
yes, when remote port is unreachable ,544 line remove operate will not execute。
but it will retry until max retry times, at last, 519 line remove operate will execute.
so it is no problem。

Reply:
Great, you are right. Another place you may need to modify is the class Receiver. Consider an exception occurred before `udpSendTimeMap.remove(ackKey);` or `AckEntry ackEntry = ackMap.remove(ackKey);`, This still triggers memory leak.